signac/contrib/migration/v0_to_v1.py

```diff
@@ -6,8 +6,61 @@
 This migration is a null-migration that serves as a template
 for future migrations and testing purposes.
 """
+import os
+import warnings
 
+from deprecation import deprecated
 
-def migrate_v0_to_v1(project):
+from ...version import __version__
+
+# A minimal v1 config.
+_cfg = """
+schema_version = string()
+"""
+
+
+def _load_config_v1(root_directory):
+    try:
+        import configobj
+    except ModuleNotFoundError:
+        try:
+            from signac.common import configobj
+        except ModuleNotFoundError:
+            raise RuntimeError(
+                "signac schema version 1 can only be read with configobj. "
+                "Please install configobj and try again."
```

Comment:
configobj hasn't been released since 2014... 😬 Should we mention "configobj or signac 1.x" since signac 1.x may be installable for longer than configobj itself? Not sure.

Reply:
Well we can't use signac 1.x because even though that could load the config file it wouldn't be able to perform the migration. If we're really worried about this one option would be to not delete the configobj embedded in signac until signac >2.0 so that a user could use signac 2.0 for the migration even if configobj was no longer installable as a standalone. I don't think we really need to worry about this though; in the worst case scenario where configobj completely vanishes (both source and distributions) we can always add a section to the docs on how to extract the source of configobj from signac 1.x and make it available for the purpose of migrations.

Reply:
I think it would be better to not change the import logic and to simply continue to include configobj with the package since it is _required_ for the migration and thus a real dependency of signac 2.0.

Reply:
I'm fine including it for a while, although I think it would be annoying to include indefinitely. If not in 2.0, I'd at least consider removing it in 3.0. I don't anticipate keeping signac _quite_ this dependency free forever. In particular, when we switch to TOML do we want to embed a TOML parsing package still, or just make one a dependency?

Reply:
I would suggest to include it as long as we support the related migration. I think it is fairly typical to expect users to install intermediate major versions for longer migration paths. This means signac v3 would only support a migration from v2, but not from v0 or v1.

Reply:
That sounds reasonable to me.